src/client/timetable/Calendar.tsx

```diff
@@ -0,0 +1,138 @@
+import React from "react";
+import {
+    startOfMonth,
+    startOfISOWeek,
+    endOfMonth,
+    endOfISOWeek,
+    eachDayOfInterval,
+    format,
+    startOfDay,
+} from "date-fns";
+import { CourseActivityResponseType } from "../../types";
+import { CalendarDay } from "./CalendarDay";
+import "./Calendar.less";
+
+interface Props {
+    sessions: Array<CourseActivityResponseType>;
+}
+
+interface State {
+    chosenMonth: number;
+    chosenYear: number;
+}
+
+export const Calendar: React.FunctionComponent<Props> = ({ sessions }) => {
+    const [{ chosenMonth, chosenYear }, setTimeFrame] = React.useState<State>({
+        chosenMonth: new Date().getMonth(),
+        chosenYear: new Date().getFullYear(),
+    });
+    const firstDayOfMonth: Date = startOfMonth(
+        new Date(chosenYear, chosenMonth)
+    );
+    const firstDayShown: Date = startOfISOWeek(firstDayOfMonth);
+    const lastDayOfMonth: Date = startOfDay(endOfMonth(firstDayOfMonth));
+    const lastDayShown: Date = startOfDay(endOfISOWeek(lastDayOfMonth));
+    const lastMonthRange: Array<Date> = eachDayOfInterval({
+        start: firstDayShown,
+        end: firstDayOfMonth,
+    }).filter((date) => date.getTime() !== firstDayOfMonth.getTime());
+    const chosenMonthRange: Array<Date> = eachDayOfInterval({
+        start: firstDayOfMonth,
+        end: lastDayOfMonth,
+    });
+    const nextMonthRange: Array<Date> = eachDayOfInterval({
+        start: lastDayOfMonth,
+        end: lastDayShown,
+    }).filter((date) => date.getTime() !== lastDayOfMonth.getTime());
+    const monthHeading: string = format(firstDayOfMonth, "MMMM, yyyy");
+    return (
```

Comment:
Consider splitting the heading to its own component as this component is fairly large.

Reply:
Done. CalendarHeader is now a separate component.